internal/venice-common/src/main/java/com/linkedin/venice/pubsub/adapter/kafka/admin/ApacheKafkaAdminAdapter.java

```diff
@@ -39,155 +40,284 @@
+      }
+      throw new PubSubClientException("Failed to get configs for kafka topic: " + pubSubTopic, e);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new PubSubClientException("Interrupted while getting configs for kafka topic: " + pubSubTopic, e);
     }
   }
 
+  /**
+   * Retrieves a set of all available PubSub topics from the Kafka cluster.
+   *
+   * @return A Set of PubSubTopic objects representing all available Kafka topics.
+   *
+   * @throws PubSubClientRetriableException If a retriable error occurs while attempting to retrieve the list of topics.
+   * @throws PubSubClientException If an error occurs while attempting to retrieve the list of topics or the current thread is interrupted while attempting to retrieve the list of topics.
+   */
   @Override
   public Set<PubSubTopic> listAllTopics() {
-    ListTopicsResult listTopicsResult = getKafkaAdminClient().listTopics();
+    ListTopicsResult listTopicsResult = internalKafkaAdminClient.listTopics();
     try {
       return listTopicsResult.names()
           .get()
           .stream()
           .map(t -> pubSubTopicRepository.getTopic(t))
           .collect(Collectors.toSet());
-    } catch (Exception e) {
-      throw new VeniceException("Failed to list all topics due to exception: ", e);
+    } catch (ExecutionException e) {
+      if (e.getCause() instanceof RetriableException) {
+        throw new PubSubClientRetriableException("Failed to list topics", e);
+      }
+      throw new PubSubClientException("Failed to list topics", e);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new PubSubClientException("Interrupted while listing topics", e);
     }
   }
 
-  // TODO: If we decide that topic deletion is always going to be blocking then we might want to get the future here and
-  // catch/extract any expected exceptions such as UnknownTopicOrPartitionException.
-  @Override
-  public KafkaFuture<Void> deleteTopic(PubSubTopic topic) {
-    return getKafkaAdminClient().deleteTopics(Collections.singleton(topic.getName())).values().get(topic.getName());
-  }
-
+  /**
+   * Sets the configuration for a Kafka topic.
+   *
+   * @param pubSubTopic The PubSubTopic for which to set the configuration.
+   * @param pubSubTopicConfiguration The configuration to be set for the specified Kafka topic.
+   * @throws PubSubTopicDoesNotExistException If the specified Kafka topic does not exist.
+   * @throws PubSubClientException If an error occurs while attempting to set the topic configuration or if the current thread is interrupted while attempting to set the topic configuration.
+   */
   @Override
-  public void setTopicConfig(PubSubTopic topic, PubSubTopicConfiguration pubSubTopicConfiguration)
-      throws TopicDoesNotExistException {
+  public void setTopicConfig(PubSubTopic pubSubTopic, PubSubTopicConfiguration pubSubTopicConfiguration)
```

Comment:
Nitpick: I don't think it adds much value to change all these `topic` variable names to `pubSubTopic`... it doesn't really add much clarity to the code (IMHO) and bloats the change set which is already fairly large. Anyway, it's not a big deal, feel free to keep it as is if you think it's better.

Reply:
I'll keep this change as for this PR and will avoid this in the future.